plugin/pkg/admission/serviceplan/changevalidator/admission.go

```diff
@@ -72,7 +72,11 @@ func (d *denyPlanChangeIfNotUpdatable) Admit(a admission.Attributes) error {
 		return apierrors.NewBadRequest("Resource was marked with kind Instance but was unable to be converted")
 	}
 
-	sc, err := d.scLister.Get(instance.Spec.ClusterServiceClassExternalName)
+	if instance.Spec.ClusterServiceClassRef == nil {
+		return nil // user chose a service class that doesn't exist
```

Comment:
Comment is slightly inaccurate: Also possibly the case where the instance wasn't yet reconciled (so its references weren't resolved) before they changed the plan. But in that case it's harmless, so still works returning `nil` here.

A little bit of a weird situation, though, because if the instance fails to provision, it will still be non-plan-updateable.